internal/endpoint/smtp/session.go

```diff
@@ -353,6 +389,12 @@ func (s *Session) prepareBody(ctx context.Context, r io.Reader) (textproto.Heade
 		}
 	}
 
+	limitr.AddLimit(s.endp.serv.MaxMessageBytes-s.endp.maxHeaderBytes, &exterrors.SMTPError{
+		Code:         552,
+		EnhancedCode: exterrors.EnhancedCode{5, 3, 4},
+		Message:      "Max message size exceeded",
+	})
+
 	buf, err := s.endp.buffer(bufr)
```

Comment:
@foxcpp I guess the current option is better. Otherwise if I disable the limit before this line, the whole message will be put into the buffer. Having a limit like this, will make the buffering fail now with the proper error.